packages/terra-date-time-picker/src/DateTimePicker.jsx

```diff
@@ -0,0 +1,361 @@
+    const isOldTimeAmbiguous = this.state.isAmbiguousTime;
+    if (dateTime && dateTime.isValid()) {
+      const tempDateTime = dateTime.clone();
+      isDateTimeAmbiguous = DateTimeUtils.checkAmbiguousTime(tempDateTime);
+    }
+
+    this.setState({
+      isAmbiguousTime: isDateTimeAmbiguous,
+      isTimeClarificationOpen: isDateTimeAmbiguous && !isOldTimeAmbiguous,
+    });
+  }
+
+  handleDateChange(event, date) {
+    if (event.type === 'change') {
+      this.dateValue = event.target.value;
+    }
+
+    let updatedDateTime;
+    const formattedDate = DateTimeUtils.formatISODateTime(date, 'YYYY-MM-DD');
+
+    if (DateTimeUtils.isValidDate(formattedDate, 'YYYY-MM-DD')) {
+      const previousDateTime = this.state.dateTime ? this.state.dateTime.clone() : null;
+      updatedDateTime = DateTimeUtils.syncDateTime(previousDateTime, date, this.timeValue);
+
+      if (DateTimeUtils.isValidTime(this.timeValue)) {
+        this.timeValue = DateTimeUtils.formatISODateTime(updatedDateTime.format(), 'HH:mm');
+      }
+    }
+
+    this.handleChange(event, updatedDateTime);
+  }
+
+  handleDateChangeRaw(event, date) {
+    this.dateValue = event.target.value;
+    this.handleChangeRaw(event, date);
+  }
+
+  handleTimeChange(event, time) {
+    this.timeValue = time;
+    const dateFormat = DateUtil.getFormatByLocale(this.context.intl.locale);
+    const validDate = DateTimeUtils.isValidDate(this.dateValue, dateFormat);
+    const validTime = DateTimeUtils.isValidTime(this.timeValue);
+    const previousDateTime = this.state.dateTime ? this.state.dateTime.clone() : null;
+
+    // If both date and time are valid, check if the time is the missing hour and invoke onChange.
+    // If the date is valid but time is invalid, the time in the dateTime state needs to be cleared and render.
+    if (validDate && validTime) {
+      const updatedDateTime = DateTimeUtils.updateTime(previousDateTime, time);
+
+      if (event.keyCode === keyCodes.ARROWDOWN &&
+        previousDateTime && updatedDateTime && previousDateTime.format() === updatedDateTime.format()) {
+        updatedDateTime.subtract(1, 'hours');
+      }
+
+      this.timeValue = DateTimeUtils.formatISODateTime(updatedDateTime.format(), 'HH:mm');
+      this.handleChangeRaw(event, this.timeValue);
+      this.handleChange(event, updatedDateTime);
+    } else {
+      // If the date is valid but the time is not, the time part in the dateTime state needs to be cleared to reflect the change.
+      if (validDate && !validTime) {
+        const updatedDateTime = DateTimeUtils.updateTime(previousDateTime, '00:00');
+
+        this.setState({
+          dateTime: updatedDateTime,
+        });
+      }
+
+      this.handleChangeRaw(event, time);
+    }
+  }
+
+  handleChange(event, newDateTime) {
+    this.setState({
+      dateTime: newDateTime,
+    });
+
+    if (this.props.onChange) {
+      this.props.onChange(event, newDateTime && newDateTime.isValid() ? newDateTime.format() : '');
```

Comment:
How will consumers of this component have to handle invalid input such as MM/DD/YY so that they can provide some feedback to a user that a value entered was invalid? Or is the expected behavior users should see is the date/time picker will clear because the input is invalid and an empty value is returned by onChange() which then tells the app to render the date/time picker with a value of ''?

Reply:
This component currently does not enforce formatting. There is an [enhancement](https://github.com/cerner/terra-core/issues/303) to enforce formatting or some sort of validation. onChange will get triggered when entry is valid. There is another prop called onChangeRaw that gets triggered on every change event.

Reply:
Thanks. That's good to know for me because that will be something needed before I think broad usage of the component will happen.